src/Controller/UserController.php

```diff
@@ -145,4 +147,59 @@ public function deleteUser($id, UserRepository $userRepository): JsonResponse
 
         return new JsonResponse(['message' => 'The user was successfully deleted!'], Response::HTTP_OK);
     }
+
+    /**
+     * Change password
+     * @Rest\Post("/{id}/change_password")
+     * @param int $id
+     * @param UserRepository $userRepository
+     * @param Request $request
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function userChangePassword($id, UserRepository $userRepository, Request $request): JsonResponse
+    {
+        $user = $userRepository->find($id);
+        if (!$user) {
+            return new JsonResponse(['message' => 'The user was not found!'], Response::HTTP_NOT_FOUND);
+        }
+        $data = $request->getContent();
```

Comment:
Add a check that you can change only your password. Please compare the logged in user with the user which is being edited.